apps/prairielearn/src/pages/enroll/enroll.html.ts

```diff
@@ -0,0 +1,282 @@
+import { z } from 'zod';
+import { html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+export const CourseInstanceSchema = z.object({
+  label: z.string(),
+  short_label: z.string(),
+  course_instance_id: z.string(),
+  enrolled: z.boolean(),
+  instructor_access: z.boolean(),
+});
+type CourseInstance = z.infer<typeof CourseInstanceSchema>;
+
+interface EnrollProps {
+  courseInstances: CourseInstance[];
+  resLocals: Record<string, any>;
+}
+
+export function Enroll({ courseInstances, resLocals }: EnrollProps) {
+  return html`
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head') %>", {
+          ...resLocals,
+          navPage: 'enroll',
+          pageTitle: 'Courses',
+        })}
+      </head>
+      <body>
+        <script>
+          $(function () {
+            $('[data-toggle="popover"]').popover({ sanitize: false });
+          });
+        </script>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'enroll',
+        })}
+        <main id="content" class="container">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white">Courses</div>
+            <table class="table table-sm table-hover table-striped">
+              <tbody>
+                ${courseInstances.map((course_instance, idx) => {
+                  return html`
+                    <tr>
+                      <td class="align-middle">${course_instance.label}</td>
+                      ${course_instance.instructor_access
+                        ? html`
+                            <td class="align-middle text-center" colspan="2">
+                              <span class="badge badge-info">instructor access</span>
+                            </td>
+                          `
+                        : html`
+                            <td>
+                              ${!course_instance.enrolled
+                                ? html`
+                                    <button
+                                      type="button"
+                                      class="btn btn-sm btn-info"
+                                      data-toggle="modal"
+                                      data-target="#addModal${idx}"
+                                    >
+                                      Add course
+                                    </button>
+                                    <div
+                                      class="modal fade"
+                                      id="addModal${idx}"
+                                      tabindex="-1"
+                                      role="dialog"
+                                      aria-labelledby="addModal${idx}Title"
+                                      aria-hidden="true"
+                                    >
```

Comment:
We really shouldn't be doing the inline-modal-per-row thing. I can make the change to a templated modal in this PR if that's desired.

Reply:
As you like. The templated modal is definitely the way to go (including pulling `Modal()` and `templateFromAttributes()` over from PT).

Reply:
I think I'll leave this as-is for now just so I can continue progress on enrollment limits and billing.

We should discuss our strategy for those sort of generic-ish bits of client-side functionality. Do we duplicate them, or put them in some kind of sharable package in PL?